goal_src/engine/gfx/generic/generic.gc

```diff
@@ -5,3 +5,150 @@
 ;; name in dgo: generic
 ;; dgos: GAME, ENGINE
 
+; This is actually defined in generic-vu1.gc but this file builds first
+(define-extern generic-init-buf (function dma-buffer int uint none))
+
+;; definition for symbol *generic-foreground-sinks*, type (array generic-dma-foreground-sink)
+(define
+  *generic-foreground-sinks*
+  (the-as (array generic-dma-foreground-sink)
+   (new
+    'static
+    'boxed-array
+    :type generic-dma-foreground-sink :length 0 :allocated-length 9
+    )
+   )
+  )
+
+;; failed to figure out what this is:
+(set!
+  (-> *generic-foreground-sinks* 0)
+  (the-as
+   generic-dma-foreground-sink
+   (-> *level* level0 foreground-sink-group 0 merc-sink)
+   )
+  )
+
+;; failed to figure out what this is:
+(set!
+  (-> *generic-foreground-sinks* 1)
+  (the-as
+   generic-dma-foreground-sink
+   (-> *level* level0 foreground-sink-group 1 merc-sink)
+   )
+  )
+
+;; failed to figure out what this is:
+(set!
+  (-> *generic-foreground-sinks* 2)
+  (the-as
+   generic-dma-foreground-sink
+   (-> *level* level1 foreground-sink-group 0 merc-sink)
+   )
+  )
+
+;; failed to figure out what this is:
+(set!
+  (-> *generic-foreground-sinks* 3)
+  (the-as
+   generic-dma-foreground-sink
+   (-> *level* level1 foreground-sink-group 1 merc-sink)
+   )
+  )
+
+;; failed to figure out what this is:
+(set!
+  (-> *generic-foreground-sinks* 4)
+  (the-as
+   generic-dma-foreground-sink
+   (-> *level* level-default foreground-sink-group 0 merc-sink)
+   )
+  )
+
+;; failed to figure out what this is:
+(set!
+  (-> *generic-foreground-sinks* 5)
+  (the-as
+   generic-dma-foreground-sink
+   (-> *level* level-default foreground-sink-group 1 merc-sink)
+   )
+  )
+
+;; failed to figure out what this is:
+(set!
+  (-> *generic-foreground-sinks* 6)
+  (new 'static 'generic-dma-foreground-sink :bucket 30)
```

Comment:
Not an issue for this PR but we may want to switch these to the `bucket-id` enum at some point.